R/tags.R

```diff
@@ -427,10 +427,24 @@ tagGetAttribute <- function(tag, attr) {
     return (NULL)
   }
 
-  # Convert all attribs to chars explicitly; prevents us from messing up factors
-  result <- lapply(attribs[attrIdx], as.character)
-  # Separate multiple attributes with the same name
-  result <- paste(result, collapse  = " ")
+  result <- attribs[attrIdx]
+  if (all(vapply(result, is.atomic, logical(1)))) {
+    # Convert all attribs to chars explicitly; prevents us from messing up factors
+    # Separate multiple attributes with the same name
+    vals <- lapply(result, function(val) {
+      val <- as.character(val)
```

Comment:
One thing that has annoyed me about how `tagGetAttribute()` currently works is this behavior:

```r
> div(foo=NA)
<div foo></div>
> tagGetAttribute(div(foo=NA), "foo")
[1] "NA"
```

Maybe it'd make since to _not_ convert to character when `isTRUE(is.na(val))`?

Reply:
Absolutely! Great catch

Reply:
Using the cran version...

```r
tagGetAttribute(div(foo=NA), "foo")
[1] "NA"
```

I still believe the change is correct.

Reply:
Addressed in 26b3042